File: frondend/product_ui/src/service/product.service.js
import axios from "axios";

const API_URL = "http://localhost:8080";

class ProductService {
    saveProduct(product) {
        return axios.post(`${API_URL}/saveProduct`, product);
    }

    editProduct(product) {
        return axios.put(`${API_URL}/editProduct/${product.id}`, product);
    }


    getAllProduct() {
        return axios.get(`${API_URL}/`);
    }

    getProductById(id) {
        return axios.get(`${API_URL}/${id}`);
    }

    deleteProduct(id) {
        return axios.delete(`${API_URL}/deleteProduct/${id}`);
    }
}

export default new ProductService();

File: backend/Project_management_system/src/main/java/com/rahul/Project_management_system/service/ProductService.java
package com.rahul.Project_management_system.service;

import java.util.List;

import com.rahul.Project_management_system.model.Product;

public interface ProductService 
{
	public Product saveProduct(Product product);
	
	public Product editProduct(Product product,Integer id);
	
	public List<Product> getAllProduct();
	
	public Product  getProductById(Integer id);
	
	public String deleteProduct(Integer id);
}
